pkg/infra/game/agent/agent.go

```diff
@@ -13,8 +12,9 @@ import (
 )
 
 type Strategy interface {
-	HandleFightMessage(m message.TaggedMessage, view *state.View, agent BaseAgent, log *immutable.Map[commons.ID, decision.FightAction]) decision.FightAction
-	Default() decision.FightAction
+	HandleFightInformation(m message.TaggedMessage, view *state.View, agent BaseAgent, log *immutable.Map[commons.ID, decision.FightAction])
+	HandleFightRequest(m message.TaggedMessage, view *state.View, log *immutable.Map[commons.ID, decision.FightAction]) message.Payload
```

Comment:
Are we going to have 2 of these methods for every type of message that needs to be sent. This seems very verbose and isn't intuitive as to what message should be sent via which method. I understand the rationale for it but is there a way we could make it more explicit somehow?